code/area.dm

```diff
@@ -249,6 +253,9 @@ ABSTRACT_TYPE(/area) // don't instantiate this directly dummies, use /area/space
 
 						//Put whatever you want here. See Entering above.
 
+		if(istype(A, /obj/critter/turtle))
+			src.sylvesters -= A
```

Comment:
absolutely not, just check if by_type[turtle].area == lightswitch.area

Reply:
Yes, I only made it like this because mbc said to, I didn't think anyone else but them would review this, I am very sorry you had to read it.